--- nmap/scan_engine.cc
+++ nmap/scan_engine.cc
@@ -4136,7 +4136,7 @@ static bool get_pcap_result(UltraScanInfo *USI, struct timeval *stime) {
       }
     } else if (ip->ip_p == IPPROTO_SCTP && !USI->prot_scan) {
       struct sctp_hdr *sctp = (struct sctp_hdr *) ((u8 *) ip + ip->ip_hl * 4);
-      struct dnet_sctp_chunkhdr *chunk = (struct dnet_sctp_chunkhdr *) ((u8 *) sctp + 12);
+      struct sctp_chunkhdr *chunk = (struct sctp_chunkhdr *) ((u8 *) sctp + 12);
       /* Now ensure this host is even in the incomplete list */
       memset(&sin, 0, sizeof(sin));
       sin.sin_addr.s_addr = ip->ip_src.s_addr;
@@ -4749,7 +4749,7 @@ static int get_ping_pcap_result(UltraScanInfo *USI, struct timeval *stime) {
       }
     } else if (ip->ip_p == IPPROTO_SCTP && USI->ptech.rawsctpscan) {
       struct sctp_hdr *sctp = (struct sctp_hdr *) (((char *) ip) + 4 * ip->ip_hl);
-      struct dnet_sctp_chunkhdr *chunk = (struct dnet_sctp_chunkhdr *) ((u8 *) sctp + 12);
+      struct sctp_chunkhdr *chunk = (struct sctp_chunkhdr *) ((u8 *) sctp + 12);
       /* Search for this host on the incomplete list */
       memset(&sin, 0, sizeof(sin));
       sin.sin_addr.s_addr = ip->ip_src.s_addr;
