--- nmap-3.10ALPHA3/configure.in.orig	2002-09-23 19:06:05 +0400
+++ nmap-3.10ALPHA3/configure.in	2002-09-23 20:12:48 +0400
@@ -5,13 +5,6 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_INIT(nmap.cc)
 
-if test -d /usr/local/lib; then
-  LDFLAGS="$LDFLAGS -L/usr/local/lib"
-fi
-if test -d /usr/local/include; then
-  CXXFLAGS="$CXXFLAGS -I/usr/local/include"
-fi
-
 libpcapdir=libpcap-possiblymodified
 AC_SUBST(libpcapdir)
 
@@ -20,9 +13,6 @@
 
 dnl Checks for programs.
 AC_PROG_CXX
- if test -n "$GCC"; then
-      CXXFLAGS="$CXXFLAGS -Wall "
- fi
 dnl AC_PROG_INSTALL
 dnl AC_PATH_PROG(MAKEDEPEND, makedepend)
 
--- nmap-3.10ALPHA3/nbase/configure.in.orig	2002-09-01 04:03:14 +0400
+++ nmap-3.10ALPHA3/nbase/configure.in	2002-09-23 20:17:40 +0400
@@ -12,22 +12,12 @@
 
 AC_INIT(nbase.h)
 
-if test -d /usr/local/lib; then
-  LDFLAGS="$LDFLAGS -L/usr/local/lib"
-fi
-if test -d /usr/local/include; then
-  CFLAGS="$CFLAGS -I/usr/local/include"
-fi
-
 dnl use config.h instad of -D macros
 AC_CONFIG_HEADER(nbase_config.h)
 
 dnl Checks for programs.
 AC_PROG_CC
 
- if test -n "$GCC"; then
-      CFLAGS="$CFLAGS -Wall "
- fi
 AC_PROG_RANLIB
 dnl AC_PROG_INSTALL
 dnl AC_PATH_PROG(MAKEDEPEND, makedepend)
--- nmap-3.10ALPHA3/nbase/configure.orig	2002-09-01 04:10:46 +0400
+++ nmap-3.10ALPHA3/nbase/configure	2002-09-23 20:38:24 +0400
@@ -1238,12 +1238,6 @@
 
 
 
-if test -d /usr/local/lib; then
-  LDFLAGS="$LDFLAGS -L/usr/local/lib"
-fi
-if test -d /usr/local/include; then
-  CFLAGS="$CFLAGS -I/usr/local/include"
-fi
 
           ac_config_headers="$ac_config_headers nbase_config.h"
 
@@ -2060,9 +2054,6 @@
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
 
 
- if test -n "$GCC"; then
-      CFLAGS="$CFLAGS -Wall "
- fi
 if test -n "$ac_tool_prefix"; then
   # Extract the first word of "${ac_tool_prefix}ranlib", so it can be a program name with args.
 set dummy ${ac_tool_prefix}ranlib; ac_word=$2
