--- nmap-4.01/nmapfe.desktop.orig	2005-04-15 01:23:33 +0000
+++ nmap-4.01/nmapfe.desktop	2006-03-03 01:13:15 +0000
@@ -1,12 +1,12 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=NmapFE
-GenericName=Security Scanner
+GenericName=Nmap Security Scanner
 Comment=GUI frontend for Nmap security/port scanner
 Version=1.0
 TryExec=nmapfe
 Exec=nmapfe
-Icon=icon-network
+Icon=nmapfe
 Terminal=false
 Type=Application
-Categories=Application;Network;System;Security;GTK
\ No newline at end of file
+Categories=Application;Network;System;Security;GTK
