--- nmap/ncat/Makefile.in
+++ nmap/ncat/Makefile.in
@@ -128,7 +128,6 @@ install: $(TARGET)
 	@echo Installing Ncat;
 	$(SHTOOL) mkdir -f -p -m 755 $(DESTDIR)$(bindir) $(DESTDIR)$(pkgdatadir) $(DESTDIR)$(mandir)/man1
 	$(INSTALL) -c -m 755 -s ncat $(DESTDIR)$(bindir)/ncat
-	$(INSTALL) -c -m 644 $(DATAFILES) $(DESTDIR)$(pkgdatadir)/
 	$(INSTALL) -c -m 644 docs/$(TARGET).1 $(DESTDIR)$(mandir)/man1/$(TARGET).1
 
 uninstall:
--- nmap/ncat/ncat_posix.c
+++ nmap/ncat/ncat_posix.c
@@ -134,28 +134,14 @@ void set_lf_mode(void)
 }
 
 #ifdef HAVE_OPENSSL
-
-#define NCAT_CA_CERTS_PATH (NCAT_DATADIR "/" NCAT_CA_CERTS_FILE)
-
 int ssl_load_default_ca_certs(SSL_CTX *ctx)
 {
-    int rc;
-
     if (o.debug)
-        logdebug("Using system default trusted CA certificates and those in %s.\n", NCAT_CA_CERTS_PATH);
+        logdebug("Using system default trusted CA certificates\n");
 
     /* Load distribution-provided defaults, if any. */
     assert(SSL_CTX_set_default_verify_paths(ctx) > 0);
 
-    /* Also load the trusted certificates we ship. */
-    rc = SSL_CTX_load_verify_locations(ctx, NCAT_CA_CERTS_PATH, NULL);
-    if (rc != 1) {
-        if (o.debug)
-            logdebug("Unable to load trusted CA certificates from %s: %s\n",
-                NCAT_CA_CERTS_PATH, ERR_error_string(ERR_get_error(), NULL));
-        return -1;
-    }
-
     return 0;
 }
 #endif
