--- nmap-4.65/configure.ac
+++ nmap-4.65/configure.ac
@@ -105,6 +105,7 @@ AC_PATH_TOOL([STRIP], [strip], [/bin/true])
 needs_cpp_precomp=no
 
 LUA_CFLAGS="-DLUA_USE_POSIX -DLUA_USE_DLOPEN"
+NSE_LDFLAGS=""
 
 case "$host" in
   *alpha-dec-osf*)
@@ -112,15 +113,15 @@ case "$host" in
     ;;
   *-netbsd* | *-knetbsd*-gnu)
     AC_DEFINE(NETBSD)
-    LDFLAGS="-Wl,-E $LDFLAGS" # needed for nse-C-module support
+    NSE_LDFLAGS="-Wl,-E" # needed for nse-C-module support
     ;;
   *-openbsd*)
     AC_DEFINE(OPENBSD)
-    LDFLAGS="-Wl,-E $LDFLAGS" # needed for nse-C-module support
+    NSE_LDFLAGS="-Wl,-E" # needed for nse-C-module support
     ;;
   *-freebsd* | *-kfreebsd*-gnu | *-dragonfly*)
     AC_DEFINE(FREEBSD)
-    LDFLAGS="-Wl,-E $LDFLAGS" # needed for nse-C-module support
+    NSE_LDFLAGS="-Wl,-E" # needed for nse-C-module support
     ;;
   *-bsdi*)
     AC_DEFINE(BSDI)
@@ -171,7 +172,7 @@ case "$host" in
   *-linux*)
     AC_DEFINE(LINUX)
     AC_DEFINE(PCAP_TIMEOUT_IGNORED)  # libpcap doesn't even LOOK at the timeout you give it under Linux
-    LDFLAGS="-Wl,-E $LDFLAGS" # needed for nse-C-module support
+    NSE_LDFLAGS="-Wl,-E" # needed for nse-C-module support
     ;;
   *-apple-darwin*)
     AC_DEFINE(MACOSX)
@@ -564,6 +565,7 @@ else
     AC_MSG_RESULT(yes) 
   fi
 
+  LDFLAGS="$NSE_LDFLAGS $LDFLAGS"
   INSTALLNSE="install-nse"
 fi
 
