diff -Nuprk.orig nmap-3.10ALPHA3.orig/configure.in nmap-3.10ALPHA3/configure.in
--- nmap-3.10ALPHA3.orig/configure.in	2002-08-25 07:16:44 +0400
+++ nmap-3.10ALPHA3/configure.in	2002-09-23 18:03:41 +0400
@@ -172,9 +172,6 @@ AC_ARG_WITH(libpcap,
     ;;
   esac]
 )
-if test $linux = yes; then
-  have_libpcap=no
-fi
 
 if test $needs_cpp_precomp = yes; then
   CXXFLAGS="-no-cpp-precomp $CXXFLAGS"
diff -Nuprk.orig nmap-3.10ALPHA3.orig/Makefile.in nmap-3.10ALPHA3/Makefile.in
--- nmap-3.10ALPHA3.orig/Makefile.in	2002-09-08 22:46:15 +0400
+++ nmap-3.10ALPHA3/Makefile.in	2002-09-23 18:03:41 +0400
@@ -14,8 +14,6 @@ deskdir = $(prefix)/share/gnome/apps/Uti
 NBASEDIR=@NBASEDIR@
 # CC = @CXX@
 CCOPT = 
-LIBPCAPDIR = @libpcapdir@
-INCLS = -I$(LIBPCAPDIR) 
 DEFS = @DEFS@ -DNMAP_VERSION=\"$(NMAP_VERSION)\" -DNMAP_NAME=\"$(NMAP_NAME)\" -DNMAP_URL=\"$(NMAP_URL)\" -DNMAP_PLATFORM=\"$(NMAP_PLATFORM)\" -DNMAPDATADIR=\"$(nmapdatadir)\"
 CXXFLAGS = -g -Wall @CXXFLAGS@ $(CCOPT) $(DEFS) $(INCLS)
 # CFLAGS = $(CXXFLAGS)
diff -Nuprk.orig nmap-3.10ALPHA3.orig/tcpip.cc nmap-3.10ALPHA3/tcpip.cc
--- nmap-3.10ALPHA3.orig/tcpip.cc	2002-09-14 08:54:46 +0400
+++ nmap-3.10ALPHA3/tcpip.cc	2002-09-23 18:04:19 +0400
@@ -190,6 +190,8 @@ pcap_t *my_pcap_open_live(char *device, 
           "*BSD:  If you are getting device not configured, you need to recompile your kernel with Berkeley Packet Filter support.  If you are getting No such file or directory, try creating the device (eg cd /dev; MAKEDEV <device>; or use mknod).\n"
           "SOLARIS:  If you are trying to scan localhost and getting '/dev/lo0: No such file or directory', complain to Sun.  I don't think Solaris can support advanced localhost scans.  You can probably use \"-P0 -sT localhost\" though.\n\n", err0r);
   }
+  if (to_ms && pcap_setnonblock(pt, 1, err0r) < 0)
+    fatal("pcap_setnonblock: %s", err0r);
   return pt;
 }
 
