--- nmap/nsock/src/configure.ac
+++ nmap/nsock/src/configure.ac
@@ -17,61 +17,61 @@ AC_CANONICAL_HOST
 
 case "$host" in
   *alpha-dec-osf*)
-    AC_DEFINE(DEC)
+    AC_DEFINE(DEC, 1, [Define to 1 for DEC])
     ;;
   *-netbsd* | *-knetbsd*-gnu)
-    AC_DEFINE(NETBSD)
+    AC_DEFINE(NETBSD, 1, [Define to 1 for NETBSD])
     ;;
   *-openbsd*)
-    AC_DEFINE(OPENBSD)
+    AC_DEFINE(OPENBSD, 1, [Define to 1 for OPENBSD])
     ;;
   *-sgi-irix5*)
-    AC_DEFINE(IRIX)
+    AC_DEFINE(IRIX, 1, [Define to 1 for IRIX])
     ;;
   *-sgi-irix6*)
-    AC_DEFINE(IRIX)
+    AC_DEFINE(IRIX, 1, [Define to 1 for IRIX])
     ;;
   *-hpux*)
-    AC_DEFINE(HPUX)
+    AC_DEFINE(HPUX, 1, [Define to 1 for HPUX])
     ;;
   *-solaris2.0*)  
-    AC_DEFINE(SOLARIS)
+    AC_DEFINE(SOLARIS, 1, [Define to 1 for SOLARIS])
     ;;
   *-solaris2.[[1-9]][[0-9]]*)
-    AC_DEFINE(SOLARIS)
+    AC_DEFINE(SOLARIS, 1, [Define to 1 for SOLARIS])
     ;;
   *-solaris2.1*)
-    AC_DEFINE(SOLARIS)
+    AC_DEFINE(SOLARIS, 1, [Define to 1 for SOLARIS])
     ;;
   *-solaris2.2*)
-    AC_DEFINE(SOLARIS)
+    AC_DEFINE(SOLARIS, 1, [Define to 1 for SOLARIS])
     ;;
   *-solaris2.3*)
-    AC_DEFINE(SOLARIS)
+    AC_DEFINE(SOLARIS, 1, [Define to 1 for SOLARIS])
     ;;
   *-solaris2.4*)
-    AC_DEFINE(SOLARIS)
+    AC_DEFINE(SOLARIS, 1, [Define to 1 for SOLARIS])
     ;;
   *-solaris2.5.1)
-    AC_DEFINE(SOLARIS)
+    AC_DEFINE(SOLARIS, 1, [Define to 1 for SOLARIS])
     ;;
   *-solaris*)
-    AC_DEFINE(SOLARIS)
+    AC_DEFINE(SOLARIS, 1, [Define to 1 for SOLARIS])
     ;;
   *-sunos4*)
-    AC_DEFINE(SUNOS)
+    AC_DEFINE(SUNOS, 1, [Define to 1 for SUNOS])
     ;;
   *-linux*)
-    AC_DEFINE(LINUX)
+    AC_DEFINE(LINUX, 1, [Define to 1 for LINUX])
     ;;
   *-freebsd* | *-kfreebsd*-gnu | *-dragonfly*)
-    AC_DEFINE(FREEBSD)
+    AC_DEFINE(FREEBSD, 1, [Define to 1 for FREEBSD])
     ;;
   *-bsdi*)
-    AC_DEFINE(BSDI)
+    AC_DEFINE(BSDI, 1, [Define to 1 for BSDI])
     ;;
   *-apple-darwin*)
-    AC_DEFINE(MACOSX)
+    AC_DEFINE(MACOSX, 1, [Define to 1 for MACOSX])
     ;;
 esac
 
@@ -104,7 +104,7 @@ AC_ARG_WITH(pcap,
   [ pcap_enabled="$withval" ])
 
 if test "$pcap_enabled" != "no"; then
-  AC_DEFINE(HAVE_PCAP)
+  AC_DEFINE(HAVE_PCAP, 1, [Define to 1 for libpcap support])
 fi
 
 dnl Checks for programs.
@@ -191,7 +191,7 @@ AC_SEARCH_LIBS(dlopen, dl)
 
 OPENSSL_LIBS=
 if test "$use_openssl" = "yes"; then
-  AC_DEFINE(HAVE_OPENSSL)
+  AC_DEFINE(HAVE_OPENSSL, 1, [Define to 1 for OpenSSL support])
   OPENSSL_LIBS="-lssl -lcrypto"
   LIBS="$LIBS $OPENSSL_LIBS"
 fi
@@ -200,7 +200,7 @@ AC_SUBST(OPENSSL_LIBS)
 
 AC_MSG_CHECKING([for SSL_set_tlsext_host_name])
 AC_TRY_LINK([#include <openssl/ssl.h>], [SSL_set_tlsext_host_name(NULL, NULL)],
-  [AC_MSG_RESULT([yes]); AC_DEFINE(HAVE_SSL_SET_TLSEXT_HOST_NAME)],
+  [AC_MSG_RESULT([yes]); AC_DEFINE(HAVE_SSL_SET_TLSEXT_HOST_NAME, 1, [Define to 1 for SSL_set_tlsext_host_name])],
   [AC_MSG_RESULT([no])])
 
 dnl Checks for typedefs, structures, and compiler characteristics.
