--- nmap/tcpip.cc
+++ nmap/tcpip.cc
@@ -3336,6 +3336,16 @@ static struct sys_route *getsysroutes_proc(FILE *routefp, int *howmany) {
         }
       }
     }
+    /* If device name in the route file does not match any interface, then
+       try to find at least an alias of interface that matches. */
+    if (i == numifaces) {
+      for(i=0; i < numifaces; i++) {
+        if (!strcmp(iface, ifaces[i].devname)) {
+          routes[numroutes].device = &ifaces[i];
+          break;
+        }
+      }
+    }
     if (i == numifaces) {
       error("Failed to find device %s which was referenced in /proc/net/route", iface);
       continue;
