diff -urk.orig nmap-3.40PVT17.orig/configure.ac nmap-3.40PVT17/configure.ac
--- nmap-3.40PVT17.orig/configure.ac	2003-09-15 01:50:24 +0400
+++ nmap-3.40PVT17/configure.ac	2003-09-15 01:50:44 +0400
@@ -271,10 +271,6 @@
   esac]
 )
 
-if test $linux = yes; then
-  have_libpcap=no
-fi
-
 if test $needs_cpp_precomp = yes; then
   CXXFLAGS="-no-cpp-precomp $CXXFLAGS"
 fi
diff -urk.orig nmap-3.40PVT17.orig/Makefile.in nmap-3.40PVT17/Makefile.in
--- nmap-3.40PVT17.orig/Makefile.in	2003-09-12 11:28:50 +0400
+++ nmap-3.40PVT17/Makefile.in	2003-09-15 01:50:44 +0400
@@ -18,7 +18,6 @@
 CCOPT = 
 LIBPCAPDIR = @libpcapdir@
 LIBPCREDIR = @LIBPCREDIR@
-INCLS = -I$(LIBPCAPDIR) 
 DEFS = @DEFS@ -DNMAP_VERSION=\"$(NMAP_VERSION)\" -DNMAP_NAME=\"$(NMAP_NAME)\" -DNMAP_URL=\"$(NMAP_URL)\" -DNMAP_PLATFORM=\"$(NMAP_PLATFORM)\" -DNMAPDATADIR=\"$(nmapdatadir)\"
 # For mtrace debugging -- see MTRACE define in main.cc for instructions
 # Should only be enabled during debugging and not in any real release.
