--- nmap/ncat/ncat_connect.c
+++ nmap/ncat/ncat_connect.c
@@ -15,7 +15,6 @@
 #include <stdlib.h>
 #include <string.h>
 #include <stdio.h>
-#include <errno.h>
 
 #ifdef HAVE_OPENSSL
 #include <openssl/ssl.h>
@@ -435,21 +434,11 @@ static void connect_evt_handler(nsock_pool nsp, nsock_event evt, void *mydata)
             nsi_delete(cs->sock_nsi, NSOCK_PENDING_NOTIFY);
         }
     } else if (status == NSE_STATUS_ERROR) {
-        if (socket_errno() == EINPROGRESS)
-            /* XXX: this is weird. errno always seems to be incorrect on ret from nsock. */
-            errno = ECONNREFUSED;
-        if (socket_errno() == EINTR) {
-            sleep(1);
-        } else if (socket_errno() != 0) {
-            if (o.verbose)
-                loguser("%s.\n", socket_strerror(socket_errno()));
-            exit(1);
-        }
-    } else if (status == NSE_STATUS_TIMEOUT){
-            errno = ETIMEDOUT;
-            if (o.verbose)
-                loguser("%s.\n", socket_strerror(socket_errno()));
-            exit(1);
+        loguser("%s.\n", socket_strerror(nse_errorcode(evt)));
+        exit(1);
+    } else if (status == NSE_STATUS_TIMEOUT) {
+        loguser("%s.\n", socket_strerror(ETIMEDOUT));
+        exit(1);
     }
 
     return;
