--- nmap-3.40PVT17/nbase/nbase.h.orig	2003-09-13 08:19:55 +0400
+++ nmap-3.40PVT17/nbase/nbase.h	2003-09-15 01:44:19 +0400
@@ -312,7 +312,7 @@
 /***************** String functions -- See nbase_str.c ******************/
 /* I modified this conditional because !@# Redhat does not easily provide
    the prototype even though the function exists */
-#if !defined(HAVE_STRCASESTR) || (defined(LINUX) && !defined(__USE_GNU) && !defined(_GNU_SOURCE))
+#if !defined(HAVE_STRCASESTR) || (defined(LINUX) && !defined(__USE_GNU))
 /* strcasestr is like strstr() except case insensitive */
 char *strcasestr(const char *haystack, const char *pneedle);
 #endif
