--- nmap-3.55/configure.ac.orig	2004-07-16 18:41:15 +0400
+++ nmap-3.55/configure.ac	2004-07-16 18:41:15 +0400
@@ -309,10 +309,6 @@ AC_ARG_WITH(libpcap,
   esac]
 )
 
-if test $linux = yes; then
-  have_libpcap=no
-fi
-
 if test $needs_cpp_precomp = yes; then
   CXXFLAGS="-no-cpp-precomp $CXXFLAGS"
 fi
--- nmap-3.55/Makefile.in.orig	2004-07-07 10:11:44 +0400
+++ nmap-3.55/Makefile.in	2004-07-16 18:41:15 +0400
@@ -18,7 +18,6 @@ CXX = @CXX@
 CCOPT = 
 LIBPCAPDIR = @libpcapdir@
 LIBPCREDIR = @LIBPCREDIR@
-INCLS = -I$(LIBPCAPDIR) 
 DEFS = @DEFS@ -DNMAP_VERSION=\"$(NMAP_VERSION)\" -DNMAP_NAME=\"$(NMAP_NAME)\" -DNMAP_URL=\"$(NMAP_URL)\" -DNMAP_PLATFORM=\"$(NMAP_PLATFORM)\" -DNMAPDATADIR=\"$(nmapdatadir)\"
 # For mtrace debugging -- see MTRACE define in main.cc for instructions
 # Should only be enabled during debugging and not in any real release.
